benchmark/single-source/Array2D.swift

```diff
@@ -27,12 +28,22 @@ public func run_Array2D(_ N: Int) {
     }
     A.append(B)
   }
-  for _ in 0..<N {
-    for i in 0 ..< 1024 {
-      for y in 0 ..< 1024 {
-        A[i][y] = A[i][y] + 1
-        A[i][y] = A[i][y] - 1
-      }
+  return A
+}()
+
```

Comment:
I wasn't sure what was the main point of this benchmark before, when I've noticed it has unstable memory consumption. From the [robust-microbench report](https://palimondo.github.io/robust-microbench/memory-use.html):
> The Array2D benchmark has significant memory use range of 7MB (3292 — 5101 pages or 12.9 MB — 19.9 MB)! It creates a 1K by 1K [[Int]], without reserving a capacity. The pure Int storage is at least 8 MB, plus there is some constant overhead per Array. I guess the variation depends on how big contiguous memory regions the allocator gets, while the arrays are growing when append is called and they sometimes need to be copied to new region. Though I’m not sure this is the point of the test and it maybe should be rewritten with reserved capacity for stable memory use.

Seeing you've taken the array initialization out into the `setUpFunction`… would you mind fixing it to use reserved capacity? How about something like this:
````swift
return Array(repeating: Array(0 ..< 1024), count: 1024)
````

Reply:
I simplified it. But note that your elegant single-liner is not equivalent to the original construction - performance wise, because it reuses the same inner array. This then triggers copies-on-write in the main benchmark loop.